src/FormCheckInput.tsx

```diff
@@ -54,6 +60,7 @@ const FormCheckInput: BsPrefixRefForwardingComponent<
       type = 'checkbox',
       isValid = false,
       isInvalid = false,
+      indeterminate = false,
```

Comment:
i think you need to distinguish between `undefined` and `false` otherwise how does removing the indeterminate state work? The ref will get removed but i don't think React will reset the attribute?

Reply:
yeah i didn't think about removing the prop, i also think we need to even go a step further and explicitly set it `element.indeterminate` to `false`, when `indeterminate=false`. otherwise things like this:

```ts
  const [isIndeterminate, setIndeterminate] = useState(true);

  return (
    <>
      <button onClick={() => setIndeterminate(!isIndeterminate)}>click</button>
          <Form.Check // prettier-ignore
            type="checkbox"
            indeterminate={isIndeterminate}
            onClick={() => setIndeterminate(!isIndeterminate)}
          />
    </>
  );
```

won't work, cause react won't remove it until we re-mount it, cause ref will persist across renders.

1. We mount Form.Check with `intermediate={true}`
2. We click on the button(setting `indermediate={false}` on the checkbox)
3. Nothing happens, cause we've already set the attr on the ref to true.

Reply:
unless the thing you have in mind accounts for that